index.js

```diff
@@ -273,7 +273,7 @@ function parse (uri, opts) {
         parsed.host = unescape(parsed.host)
       }
       if (parsed.path !== undefined && parsed.path.length) {
-        parsed.path = encodeURI(parsed.path)
+        parsed.path = escape(unescape(parsed.path))
```

Comment:
Could you elaborate this change?

These functions are deprecated:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI?retiredLocale=it
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/unescape

```
> escape(unescape('шеллы'))
'%u0448%u0435%u043B%u043B%u044B'
> encodeURI('шеллы')
'%D1%88%D0%B5%D0%BB%D0%BB%D1%8B'
```

Reply:
Hey,

So I want to start by saying that I fully agree with you in that deprecated functions should be avoided in general, but they have different behavior as you pointed out and this change is to achieve url-js compatibility (one of the goals of this library) — you can see that escape and unescape have already been used many times in this file:

https://github.com/fastify/fast-uri/blob/bf03d44c2888328643021e1f0ec87b08ede659e8/index.js#L265-L274

In fact, if I recall correctly, someone tried to change these functions to encode/decodeURI, and maybe the tests didn't catch the behavior change for `path`

Having said all that, I don't think these functions will ever be removed from the Web